custom_components/foxess_modbus/entities/modbus_sensor.py

```diff
@@ -60,11 +63,19 @@ def __init__(
     def native_value(self):
         """Return the value reported by the sensor."""
         value = self._controller.read(self.entity_description.address)
+
         if value is not None:
             if self.entity_description.scale is not None:
                 value = value * self.entity_description.scale
             if self.entity_description.post_process is not None:
-                return self.entity_description.post_process(value)
+                value = self.entity_description.post_process(value)
+
+            rules = self.entity_description.validate
+            if not self._validate(rules, value):
+                _LOGGER.warning(
```

Comment:
Can we log the original value read from the register, as well as the address? Just in case there's a problem with our conversion.